lab3/src/MyScanner.java

```diff
@@ -0,0 +1,213 @@
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        separators.add("\t");
+        for (int i = 0; i <= lines.size(); i++) {
+            if (i <= 10) {
+                operators.add(lines.get(i));
+            }
+            if (i > 10 && i <= 18) {
+                separators.add(lines.get(i));
+            }
+            if (i > 18 && i <= 32) {
+                reservedWords.add(lines.get(i));
+            }
+
+        }
+    }
+
+    private List<String> readFile(String filename) {
+        //read every file and return a list with the lines
+        List<String> lines = new ArrayList<>();
+        BufferedReader reader;
+        try {
+            reader = new BufferedReader(new FileReader(filename));
+            String line = reader.readLine();
+            while (line != null) {
+                lines.add(line);
+                line = reader.readLine();
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return lines;
+    }
+
+    public List<String> tokenize(String line, int lineCount) {
+        List<String> tokens = new ArrayList<>();
+        StringTokenizer st = new StringTokenizer(line, " ,()[]{};+-*/%=><!\t", true);
+        while (st.hasMoreTokens()) {
+            String token = st.nextToken();
+            if (!token.equals(" ")) {
+                if (token.equals("<") || token.equals(">")) {
+                    String nextToken = st.nextToken();
+                    if (nextToken.equals("=")) {
+                        tokens.add(token + nextToken);
+                    } else {
+                        tokens.add(token);
+                        tokens.add(nextToken);
+                    }
+                } else {
+                    if (token.equals(";")) {
+                        try {
+                            String nextToken = st.nextToken();
+                            if (token.equals(nextToken)) {
+                                tokens.add(token + nextToken);
+                            } else {
+                                tokens.add(token);
+                                tokens.add(nextToken);
+                            }
+                        } catch (NoSuchElementException e) {
+                            System.out.println("Lexical error for token: " + token + " at line: " + lineCount);
+                        }
+                    } else {
+                        tokens.add(token);
+                    }
+
+                }
+            }
+        }
+
+
+        return tokens;
+    }
+
+    private void writeToFilePif(String filename, PIF pif) throws IOException {
+        BufferedWriter writer = new BufferedWriter(new FileWriter(filename, true));
+
+        List<String> tokens = pif.getTokens();
+        List<Pair> entries = pif.getEntries();
+        writer.write(pif.toString());
+        writer.close();
+    }
+
```

Comment:
here same method could be used for writing to both the pif file and the symbol table file